packages/teams-js/src/internal/communication.ts

```diff
@@ -320,6 +324,45 @@ export function sendMessageToParentAsync<T>(actionName: string, args: any[] | un
   });
 }
 
+/**
+ * @hidden
+ * Send a message to parent requesting a MessageChannel Port.
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function requestPortFromParentWithVersion(
+  apiVersionTag: string,
+  actionName: string,
+  args: any[] | undefined = undefined,
+): Promise<MessagePort> {
+  if (!isFollowingApiVersionTagFormat(apiVersionTag)) {
+    throw Error(
+      `apiVersionTag: ${apiVersionTag} passed in doesn't follow the pattern starting with 'v' followed by digits, then underscore with words, please check.`,
+    );
+  }
+  return new Promise((resolve) => {
+    const request = sendMessageToParentHelper(apiVersionTag, actionName, args);
+    resolve(waitForPort(request.id));
+  });
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function waitForPort(requestId: number): Promise<MessagePort> {
+  return new Promise<MessagePort>((resolve, reject) => {
+    CommunicationPrivate.portCallbacks[requestId] = (port: MessagePort | undefined, args?: any[]) => {
```

Comment:
>any [](http://example.com/codeflow?start=91&length=3)

Could this be `unknown` instead of `any`? Since you aren't actually making any assumptions about the type in your code it doesn't matter in practice today, but it would make it easier for us to eventually enable the `no-explicit-any` rule in the future.  😜

Reply:
That seems to work (args is always typed any[] elsewhere in this file, so you will still have more work to do :))